sources/PreferencePanel.m

```diff
@@ -139,24 +140,33 @@ @implementation PreferencePanel {
     NSSize _standardSize;
 }
 
+static PreferencePanel *_sessionsInstance;
+static PreferencePanel *_sharedInstance;
+
 + (instancetype)sharedInstance {
-    static id instance;
-    static dispatch_once_t once;
-    dispatch_once(&once, ^{
-        instance = [[self alloc] initWithProfileModel:[ProfileModel sharedInstance]
-                               editCurrentSessionMode:NO];
-    });
-    return instance;
+    if (!_sharedInstance) {
+        _sharedInstance = [[self alloc] initWithProfileModel:[ProfileModel sharedInstance]
+                                        editCurrentSessionMode:YES];
+    }
+    return _sharedInstance;
 }
 
 + (instancetype)sessionsInstance {
-    static id instance;
-    static dispatch_once_t once;
-    dispatch_once(&once, ^{
-        instance = [[self alloc] initWithProfileModel:[ProfileModel sessionsInstance]
-                               editCurrentSessionMode:YES];
-    });
-    return instance;
+    if (!_sessionsInstance) {
+        _sessionsInstance = [[self alloc] initWithProfileModel:[ProfileModel sessionsInstance]
+                                        editCurrentSessionMode:YES];
+    }
+    return _sessionsInstance;
+}
+
++ (void)_releaseAndNilifySingletonInstancesIfNeeded:(PreferencePanel *)instance {
```

Comment:
It's pretty weird to release a singleton. This only gets you back about 30-50 megs of ram, so I'm not sure it's worth the added complexity.